week10/ns/ns.go

```diff
@@ -0,0 +1,95 @@
+package ns
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"sync"
+)
+
+type Ns struct {
```

Comment:
You might want to give the most important type in your package a slightly better name. :) For example, `ns.Service` or `ns.News`, etc...